### ISSUE TITLE
Current design has "quirks" that make validation scores somewhat invalid

### ISSUE THREAD

**Issue by Vectorrent** [0-5788719150923125/praxis#16]
It is common for training and inference to have separate and distinct behaviors. For example, during training, you might use dropout - but during inference, you disable it. This is why Pytorch provide an internal flag (`self.training`) - allowing a developer to respond to differences such as this.

In two different places, we have code where such a distinction could affect validation losses in a negative way.

~~[In the first](https://github.com/0-5788719150923125/praxis/blob/main/praxis/modules/router.py), we have a router that checks for `self.training or batch_size > 1`. This will essentially trigger validation batches to behave like training batches (using top-k instead of sigmoid), because of their batch size.~~

[In the second](https://github.com/0-5788719150923125/praxis/blob/main/praxis/modules/controller.py), we have a controller that must predict the optimal next-expert to select, during inference. However, during validation - this could would predict `batch_size` number of experts, instead of just 1. To handle this, we use `torch.mode()` - to reduce the prediction to just the most-common prediction. However, this will inevitably leave many examples in the validation batch routing through sub-optimal experts. 

I can envision ways to resolve both issues, but because of the complexity - I haven't attempted to, yet. For now, we should expect that validation losses are probably higher than they otherwise would be, with proper code.

**Comment by Vectorrent** [comment]
I was able to greatly simplify the router code, to essentially function identically in both training and inference modes.

The controller could still be an issue, though for different reasons now. It has gone through some massive revisions, and will likely go through more before we land on a final version.

**Comment by Vectorrent** [comment]
This is really not an issue anymore. We don't even use these controllers currently, and when we do - it is evident that they are not functionally complete. There is no reason to keep this one open.
